public/app/features/admin/migrate-to-cloud/onprem/EmptyState/CallToAction/ConnectModal.tsx

```diff
@@ -0,0 +1,127 @@
+            <TextLink href="https://grafana.com/auth/sign-in/" external>
+              {t('migrate-to-cloud.connect-modal.body-view-stacks', 'View my cloud stacks')}
+            </TextLink>
+            <Trans i18nKey="migrate-to-cloud.connect-modal.body-paste-stack">
+              Once you&apos;ve decided on a stack, paste the URL below.
+            </Trans>
+            <Field
+              className={styles.field}
+              invalid={!!errors.stackURL}
+              error={errors.stackURL?.message}
+              label={t('migrate-to-cloud.connect-modal.body-url-field', 'Cloud stack URL')}
+            >
+              <Input
+                {...register('stackURL', {
+                  required: t('migrate-to-cloud.connect-modal.stack-required-error', 'Stack URL is required'),
+                })}
+                id={cloudStackId}
+                placeholder="https://example.grafana.net/"
+              />
+            </Field>
+            <span>
+              <Trans i18nKey="migrate-to-cloud.connect-modal.body-token">
+                Your self-managed Grafana installation needs special access to securely migrate content. You&apos;ll
+                need to create a migration token on your chosen cloud stack.
+              </Trans>
+            </span>
+            <span>
+              <Trans i18nKey="migrate-to-cloud.connect-modal.body-token-instructions">
+                Log into your cloud stack and navigate to Administration, General, Migrate to Grafana Cloud. Create a
+                migration token on that screen and paste the token here.
+              </Trans>
+            </span>
+            <Field
+              className={styles.field}
+              invalid={!!errors.token}
+              error={errors.token?.message}
+              label={t('migrate-to-cloud.connect-modal.body-token-field', 'Migration token')}
+            >
+              <Input
+                {...register('token', {
+                  required: t('migrate-to-cloud.connect-modal.token-required-error', 'Migration token is required'),
+                })}
+                id={tokenId}
+                placeholder={t('migrate-to-cloud.connect-modal.body-token-field-placeholder', 'Paste token here')}
+              />
+            </Field>
+          </Stack>
+        </Text>
+        <Modal.ButtonRow>
+          <Button variant="secondary" onClick={hideModal}>
+            <Trans i18nKey="migrate-to-cloud.connect-modal.cancel">Cancel</Trans>
+          </Button>
+          <Button type="submit" disabled={isConnecting || !(stackURL && token)}>
+            {isConnecting
+              ? t('migrate-to-cloud.connect-modal.connecting', 'Connecting to this stack...')
+              : t('migrate-to-cloud.connect-modal.connect', 'Connect to this stack')}
+          </Button>
+        </Modal.ButtonRow>
+      </form>
+    </Modal>
+  );
+};
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  field: css({
+    alignSelf: 'stretch',
+  }),
+});
```

Comment:
You could wrap the two fields in Stack with direction column instead of using the styles.

Reply:
i don't think that works when nested in the parent `Stack`, so would have to split the existing `Stack` element which then would break the overall spacing between fields... idk, this feels like the least bad option 😅

Reply:
Why doesn't it work? I've tried with this layout and seemed fine to me: 
```tsx
          <Stack direction="column" gap={2} alignItems="flex-start">
            <Trans i18nKey="migrate-to-cloud.connect-modal.body-get-started">
              To get started, you&apos;ll need a Grafana.com account.
            </Trans>
            <TextLink href="https://grafana.com/auth/sign-up/create-user?pg=prod-cloud" external>
              {t('migrate-to-cloud.connect-modal.body-sign-up', 'Sign up for a Grafana.com account')}
            </TextLink>
            <Trans i18nKey="migrate-to-cloud.connect-modal.body-cloud-stack">
              You&apos;ll also need a cloud stack. If you just signed up, we&apos;ll automatically create your first
              stack. If you have an account, you&apos;ll need to select or create a stack.
            </Trans>
            <TextLink href="https://grafana.com/auth/sign-in/" external>
              {t('migrate-to-cloud.connect-modal.body-view-stacks', 'View my cloud stacks')}
            </TextLink>
            <Trans i18nKey="migrate-to-cloud.connect-modal.body-paste-stack">
              Once you&apos;ve decided on a stack, paste the URL below.
            </Trans>
            <Stack direction={'column'}>
              <Field
                className={styles.field}
                invalid={!!errors.stackURL}
                error={errors.stackURL?.message}
                label={t('migrate-to-cloud.connect-modal.body-url-field', 'Cloud stack URL')}
              >
                <Input
                  {...register('stackURL', {
                    required: t('migrate-to-cloud.connect-modal.stack-required-error', 'Stack URL is required'),
                  })}
                  id={cloudStackId}
                  placeholder="https://example.grafana.net/"
                />
              </Field>
              <span>
                <Trans i18nKey="migrate-to-cloud.connect-modal.body-token">
                  Your self-managed Grafana installation needs special access to securely migrate content. You&apos;ll
                  need to create a migration token on your chosen cloud stack.
                </Trans>
              </span>
              <span>
                <Trans i18nKey="migrate-to-cloud.connect-modal.body-token-instructions">
                  Log into your cloud stack and navigate to Administration, General, Migrate to Grafana Cloud. Create a
                  migration token on that screen and paste the token here.
                </Trans>
              </span>
              <Field
                required
                className={styles.field}
                invalid={!!errors.token}
                error={errors.token?.message}
                label={t('migrate-to-cloud.connect-modal.body-token-field', 'Migration token')}
              >
                <Input
                  {...register('token', {
                    required: t('migrate-to-cloud.connect-modal.token-required-error', 'Migration token is required'),
                  })}
                  id={tokenId}
                  placeholder={t('migrate-to-cloud.connect-modal.body-token-field-placeholder', 'Paste token here')}
                />
              </Field>
            </Stack>
          </Stack>
```
You basically create another div with the column direction and stretch alignment.

Reply:
but the width of that inner `Stack` is not stretching to fill, it's just based on the content inside.

that feels very brittle. if we ever change the text to be shorter it'll break:
![image](https://github.com/grafana/grafana/assets/20999846/3076757b-c803-497f-9f93-52e0a4157ad9)

Reply:
then the click target on the links span the whole modal

Reply:
Ok, I give up then 😆